src/components/qas/questions/qaCalculator/qaCalculatorUtilities.js

```diff
@@ -64,5 +122,180 @@ export const calculateCraterDiameter = ({
+    },
+    {
+      id: 'X',
+      range: [8],
+      description:
+        'Most masonry and frame structures destroyed with their foundations. Some well-built wooden structures and bridges destroyed. Serious damage to dams, dikes, and embankments. Large landslides. Water thrown on banks of canals, rivers, lakes, etc. Sand and mud shifted horizontally on beaches and flat land. Rails bent slightly.',
+    },
+    {
+      id: 'XI',
+      range: [9],
+      description:
+        'As X. Rails bent greatly. Underground pipelines completely out of service.',
+    },
+    {
+      id: 'XII',
+      range: [9, '+'],
+      description:
+        'As X. Damage nearly total. Large rock masses displaced. Lines of sight and level distorted. Objects thrown into the air.',
+    },
+  ];
+
+  return mercalliIntensity.filter(mi =>
+    includes(mi.range, Math.round(richterMagnitude))
+  );
+};
+
+export const getAirBlastDamage = airBlastOverPressure => {
+  const airBlastDamage = [
+    {
+      overPressure: 426000,
+      description:
+        'Cars and trucks will be largely displaced and grossly distorted and will require rebuilding before use.',
+    },
+    {
+      overPressure: 379000,
+      description: 'Highway girder bridges will collapse.',
+    },
+    {
+      overPressure: 297000,
+      description:
+        'Cars and trucks will be overturned and displaced, requiring major repairs',
+    },
+    {
+      overPressure: 273000,
+      description:
+        'Multistory steel-framed office-type buildings will suffer extreme frame distortion, incipient collapse',
+    },
+    {
+      overPressure: 121000,
+      description: 'Highway truss bridges will collapse.',
+    },
+    {
+      overPressure: 100000,
+      description:
+        'Highway truss bridges will suffer substantial distortion of bracing',
+    },
+    {
+      overPressure: 42600,
+      description: 'Multistory wall-bearing buildings will collapse.',
+    },
+    {
+      overPressure: 38500,
+      description:
+        'Multistory wall-bearing buildings will experience severe cracking and interior partitions will be blown down.',
+    },
+    {
+      overPressure: 26800,
+      description: 'Wood frame buildings will almost completely collapse.',
+    },
+    {
+      overPressure: 22900,
+      description:
+        'Interior partitions of wood frame buildings will be blown down. Roof will be severely damaged.',
+    },
+    {
+      overPressure: 6900,
+      description: 'Glass windows shatter.',
+    },
+  ];
+  const filteredAirBlastDamage = airBlastDamage.filter(damage => {
+    const { overPressure } = damage;
+    return overPressure <= airBlastOverPressure;
+  });
+
+  return filteredAirBlastDamage;
+};
+
+export const calculateAsteroidImpact = props => {
+  // console.log({ ...props });
+  const craterDiameter = calculateCraterDiameter(props);
```

Comment:
This is a neat trick with feeding in `props` and then spreading on the other side.  We don't make use of this trick too often (outside the obvious use in writing React components), and the reason for that is it can make it kinda challenging to trace relationships/interdependencies between methods.  I think your use here is shy of that line and A-ok, but I wouldn't recommend using this as a go to strat unless the method your giving props to spreads and then uses most of the props.  If you're spreading just to grab one or two things...well, maybe not totally justifiable.

Reply:
Okay. I believe I did this just to get going. Felt smooth enough, but I do understand your point. I went ahead and updated all them to be explicit.